WordPress/Classes/Services/PageCoordinator.swift

```diff
@@ -0,0 +1,25 @@
+import Foundation
+
+class PageCoordinator {
+    typealias TemplateSelectionCompletion = (String?) -> Void
+
+    static func showLayoutPickerIfNeeded(from controller: UIViewController, forBlog blog: Blog, completion: @escaping TemplateSelectionCompletion) {
+        if FeatureFlag.gutenbergModalLayoutPicker.enabled && blog.isGutenbergEnabled {
+            showLayoutPicker(from: controller, completion)
+        } else {
+            completion(nil)
+        }
+    }
+
+    private static func showLayoutPicker(from controller: UIViewController, _ completion: @escaping TemplateSelectionCompletion) {
+        let storyboard = UIStoryboard(name: "LayoutPickerStoryboard", bundle: Bundle.main)
```

Comment:
> both the custom Navigation Controller and the Layout picker can also be used separately if needed in the future.

This would still be possible in one storyboard. The Navigation Controller can still be created programmatically with the standard UINavigationController init methods. I can add a `storyboardID` to the Gutenberg Layout Picker View Controller Scene to allow the instantiation of that separately if needed. 

> Alternatively, we could also instantiate the NavigationController in code and set its root controller

I'm happy to switch this over if that's our standard practice. My goal here was to follow Apple's recommendation that the storyboard contains all of the design-time elements that are related to the flow of a set of screens. `GutenbergLayoutPickerViewController` assumes that the navigation controller is an instance of `GutenbergLightNavigationController` to hide the bar's shadow properly. Although with the changes I'll be making to add the visual effects @iamthomasbishop requested, I'll hide the navigation bar for this view instead. However, `GutenbergLightNavigationController` does still have other considerations in it to enforce the flow displays correctly, and using the view without a navigation controller won't work as we build out the [preview button](https://github.com/wordpress-mobile/gutenberg-mobile/issues/2452) which will push a new view controller.

Since this felt more connected in the flow of the UI I felt the storyboard should include it to provide the full picture. If that time came when we did want to reuse the root view controller, that would still be doable with minimal code changes. Being forced to make those changes would also prompt the future developer to consider the best way to handle the design settings of the navigation controller to accommodate all of the uses.

Ultimately I think either approach will work, I just wanted to share my thoughts on it, and I'm happy to proceed in whichever way you feel is best and most consistent.

Reply:
Thanks for the detail explanation!

Originally the main idea behind my suggestion was to hide the required init steps and configuration inside the `GutenbergLayoutPickerViewController`, in a way that `PageCoordinator.swift` doesn't need to know about storyboard details that seems internal to the controller (specially with it being tightly bounded with the navigation controller).

The Storyboard config as you explained it makes perfect sense to me! Let's keep it this way.